stock_picking_auto_create_lot_qty/models/stock_picking.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (C) 2023 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def _set_auto_lot(self):
```

Comment:
I think this function is too complex!
Could split method logic in separate methods?

Reply:
I'll fix it